.compatibility_tests/compatibility_test_1_15/tests/integration/compatibility_1_15.rs

```diff
@@ -0,0 +1,208 @@
+#[test_case(test_roqoqo_1_15::operations::InvSqrtPauliY::new(0).into(); "InvSqrtPauliY")]
+// Operations from 1.16
+// TODO: add the new gates
+fn test_bincode_compatibility_1_15(operation: test_roqoqo_1_15::operations::Operation) {
+    let mut test_circuit = test_roqoqo_1_15::Circuit::new();
+    test_circuit += operation;
+
+    let test_measurement_input = test_roqoqo_1_15::measurements::PauliZProductInput::new(3, false);
+    let test_measurement = test_roqoqo_1_15::measurements::PauliZProduct {
+        constant_circuit: Some(test_circuit.clone()),
+        circuits: vec![test_circuit],
+        input: test_measurement_input,
+    };
+    let test_program = test_roqoqo_1_15::QuantumProgram::PauliZProduct {
+        measurement: test_measurement,
+        input_parameter_names: vec!["test".to_string()],
+    };
+    let test_serialisation: Vec<u8> = bincode::serialize(&test_program).unwrap();
+
+    let _test_deserialisation: roqoqo::QuantumProgram =
+        bincode::deserialize(&test_serialisation).unwrap();
+}
+
+#[test]
+fn test_device_compat() {
+    let test_device = test_roqoqo_1_15::devices::AllToAllDevice::new(
+        3,
+        &["RotateZ".to_string()],
+        &["CNOT".to_string()],
+        1.0,
+    );
+    let test_serialisation: Vec<u8> = bincode::serialize(&test_device).unwrap();
+
+    let test_deserialisation: roqoqo::devices::AllToAllDevice =
+        bincode::deserialize(&test_serialisation).unwrap();
+
+    let comparsion_device = roqoqo::devices::AllToAllDevice::new(
+        3,
+        &["RotateZ".to_string()],
+        &["CNOT".to_string()],
+        1.0,
+    );
+    assert_eq!(test_deserialisation, comparsion_device);
+}
+
+// Operations from 1.11 - ApplyConstantSpinHamiltonian and ApplyTimeDependentHamiltonian are unstable in 1.11
+// use struqture;
+// use struqture::prelude::*;
+// fn create_apply_constant_spin_hamiltonian(
+// ) -> test_roqoqo_1_15::operations::ApplyConstantSpinHamiltonian {
+//     let pp = struqture::spins::PauliProduct::new().z(0);
+//     let mut hamiltonian = struqture::spins::SpinHamiltonian::new();
+//     hamiltonian
+//         .add_operator_product(pp.clone(), 1.0.into())
+//         .unwrap();
+//     return test_roqoqo_1_15::operations::ApplyConstantSpinHamiltonian::new(
+//         hamiltonian,
+//         1.0.into(),
+//     );
+// }
+
+// fn create_apply_timedependent_spin_hamiltonian(
+// ) -> test_roqoqo_1_15::operations::ApplyTimeDependentSpinHamiltonian {
+//     let pp = struqture::spins::PauliProduct::new().z(0);
+//     let mut hamiltonian = struqture::spins::SpinHamiltonian::new();
+//     hamiltonian
+//         .add_operator_product(pp.clone(), "omega".into())
+//         .unwrap();
+//     let mut values = HashMap::new();
+//     values.insert("omega".to_string(), vec![1.0]);
+//     return test_roqoqo_1_15::operations::ApplyTimeDependentSpinHamiltonian::new(
+//         hamiltonian,
+//         vec![1.0],
+//         values.clone(),
+//     );
+// }
```

Comment:
What is the plan with these commented out functions? Are ApplyConstantSpinHamiltonian and ApplyTimeDependentHamiltonian stable now?

Reply:
It seems they are not stable yet. Maybe @kbarkhqs knows better though.

Reply:
They haven't been stabilised yet, so we can't uncomment them. We can revisit them in a meeting soon